testing-files/faultyThing.js

```diff
@@ -201,23 +128,98 @@ srv.start()
+
+            thing.setActionHandler("setCounter", async (input) => {
+                console.log("* ACTION HANDLER FUNCTION for setCounter")
+                console.log("* ", input)
+                thing.writeProperty("counter", input)
+                return "not the expected return value"
+            })
+
+            thing.setActionHandler("getTemperature", () => {
+                console.log("* ACTION HANDLER FUNCTION for getTemp")
+                thing
+                    .readProperty("temperature")
+                    .then((temp) => {
+                        console.log("* getTemperature successful")
+                        return temp
+                    })
+                    .catch(() => {
+                        console.log("* getTemperature failed")
+                        return 0
+                    })
+            })
+
+            thing.setActionHandler("setDisplay", (input) => {
+                console.log("* ACTION HANDLER FUNCTION for setDisplay")
+                console.log("* ", input)
+                return new Promise((resolve, reject) => {
+                    resolve("Display set")
+                })
+            })
+
+            thing.setActionHandler("setTestObject", (input) => {
+                console.log("* ACTION HANDLER FUNCTION for setTestObject")
+                console.log("* ", input)
+                return thing.writeProperty("wrongDataTypeObject", input).then(
+                    () => input,
+                    () => false
+                )
+            })
+
+            thing.setActionHandler("longTakingAction", (input) => {
                 console.log("* ACTION HANDLER FUNCTION for longTakingAction")
-                var promise1 = new Promise(function (resolve, reject) {
+                var promise1 = new Promise((resolve, reject) => {
                     setTimeout(resolve, 5000, input)
                 })
                 return promise1
-            }
-        )
+            })
 
-        thing.addEvent("onChange", {
-            type: "number",
-        })
+            setInterval(() => {
+                thing.emitEvent("failEvent", "not a number so test fails")
+            }, 400)
+
+            setInterval(() => {
+                thing.writeProperty("counter", true)
+                thing.writeProperty("faultyPercent", 300)
+                setTimeout(async () => {
+                    await thing.writeProperty("counter", "not a number")
+                    await thing.writeProperty("faultyPercent", -400)
+                    return
+                }, 400)
+            }, 800)
 
-        thing.expose().then(() => {
-            console.info(thing.title + " ready")
+            thing.expose().then(() => {
+                console.info(thing.title + " ready")
+            })
         })
-    })
-    .catch((err) => {
-        throw "Couldnt connect to servient"
-    })
+        .catch((err) => {
+            throw "Could not connect to servient"
+        })
+})
```

Comment:
@egekorkan The faultyThing is no ready in my opinion. I added three observable properties one alternates between being set violate minimum and maximum, one emits wrong data type and one emits nothing at all. Additionally I added an event that emits the wrong data Type. Is this sufficient?

Reply:
Yes, good work!